src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java

```diff
@@ -65,7 +65,7 @@ protected CreateSnapshotResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(CreateSnapshotRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
Bottom line: I am fine with not addressing #8102 on this PR, we can have another one afterwards, I would just love to have that issue fixed soon too, regardless of whether we move snapshots to customs (which seems a bigger change).